packages/firebase/externs/firebase-auth-externs.js

```diff
@@ -1551,7 +1694,40 @@ firebase.auth.ConfirmationResult.prototype.confirm = function(
 ) {};
 
 /**
- * Asynchronously signs in as an anonymous user.
+ * Signs in a user anonymously and returns any additional user info data or
+ * credentials.
+ *
+ * If there is already an anonymous user signed in, that user with
+ * additional date will be returned; otherwise, a new anonymous user
+ * identity will be created and returned.
+ *
+ * <h4>Error Codes</h4>
+ * <dl>
+ * <dt>auth/operation-not-allowed</dt>
+ * <dd>Thrown if anonymous accounts are not enabled. Enable anonymous accounts
+ *     in the Firebase Console, under the Auth tab.</dd>
+ * </dl>
+ *
+ * @example
+ * firebase.auth().signInAnonymouslyAndRetrieveData().catch(function(error) {
+ *   // Handle Errors here.
+ *   var errorCode = error.code;
+ *   var errorMessage = error.message;
+ *
+ *   if (errorCode === 'auth/operation-not-allowed') {
+ *     alert('You must enable Anonymous auth in the Firebase Console.');
+ *   } else {
+ *     console.error(error);
+ *   }
+ * });
+ *
+ * @return {!firebase.Promise<!firebase.auth.UserCredential>}
+ */
+firebase.auth.Auth.prototype.signInAnonymouslyAndRetrieveData = function() {};
+
+/**
+ * Asynchronously signs in as an anonymous user. Will be depercated, replaced
```

Comment:
This method will be deprecated and will be updated to resolve with a `firebase.auth.UserCredential` as is returned in {@link firebase.auth.Auth#signInAnonymouslyAndRetrieveData}.